File: src/weeklyContest/weeklyContest309/Q3.java
package weeklyContest.weeklyContest309;

public class Q3 {
    public int longestNiceSubarray(int[] nums) {
        int n = nums.length, i = 0, j = 1, max = 1;
        while (j < n) {
            boolean flag = true;
            for (int k = i; k < j; k++) {
                if ((nums[k] & nums[j]) != 0) {
                    flag = false;
                    break;
                }
            }
            if (flag) {
                max = Math.max(j - i + 1, max);
            } else {
                i++;
                j--;
            }
            j++;
        }
        return max;
    }
}

File: contest/w403/q3.py
from functools import cache
from math import inf
from typing import List


class Solution:
    def maximumTotalCost(self, nums: List[int]) -> int:
        m = len(nums)

        @cache
        def dfs(i: int) -> int:
            if i < 0:
                return 0
            x = -inf
            if i > 0:
                x = dfs(i - 2) + nums[i - 1] - nums[i]
            return max(dfs(i - 1) + nums[i], x)

        return dfs(m - 1)


if __name__ == '__main__':
    nums = [-937]
    res = Solution().maximumTotalCost(nums)
    print(res)
